sdk/spring/azure-spring-integration-servicebus/src/main/java/com/azure/spring/integration/servicebus/converter/ServiceBusMessageConverter.java

```diff
@@ -73,32 +77,56 @@ protected ServiceBusMessage fromByte(byte[] payload) {
     @Override
     protected void setCustomHeaders(MessageHeaders headers, ServiceBusMessage message) {
 
+        Map<String, Object> headersForApplicationProperties = new HashMap<String, Object>();
+        headersForApplicationProperties.putAll(headers);
+
         // Spring MessageHeaders
-        getStringHeader(headers, MessageHeaders.ID).ifPresent(message::setMessageId);
```

Comment:
How about we make a copy of the messageHeaders and create a function called getAndRemove which will directly remove from the copy headers, so we don't need to keep two copies.

Reply:
Done